server/graphql/mutations/paymentMethods.js

```diff
@@ -54,18 +54,23 @@ async function createVirtualPaymentMethod(args, remoteUser) {
  * @param {email} args.email The email of the user claiming the virtual card
  * @returns {models.PaymentMethod} return the virtual card payment method.
  */
-export async function claimVirtualCard(args, remoteUser) {
+export async function claimPaymentMethod(args, remoteUser) {
   const paymentMethod = await virtualcard.claim(args, remoteUser);
   const user = await models.User.findOne({ where: { CollectiveId: paymentMethod.CollectiveId }});
-  const amount = paymentMethod.initialBalance;
-  const currency = paymentMethod.currency;
+  const {
+    initialBalance,
+    currency,
+    name
+   } = paymentMethod;
+
   const emitter = await models.Collective.findById(paymentMethod.sourcePaymentMethod.CollectiveId);
   const userCollective = await user.getCollective();
-  const name = userCollective.name && userCollective.name.match(/anonymous/i) ? '' : user.name;
+  const userName = userCollective.name && userCollective.name.match(/anonymous/i) ? '' : userCollective.name;
 
-  emailLib.send('user.card.claimed', args.email, {
-    loginLink: user.generateLoginLink(`/redeemed?name=${name}&amount=${amount}&currency=${currency}&emitterSlug=${emitter.slug}&emitterName=${emitter.name}`),
-    amount,
+  emailLib.send('user.card.claimed', user.email, {
+    loginLink: user.generateLoginLink(`/redeemed?name=${encodeURIComponent(userName)}&amount=${initialBalance}&currency=${currency}&emitterSlug=${emitter.slug}&emitterName=${encodeURIComponent(emitter.name)}`),
```

Comment:
I'm still concerned about passing display values in the URL, let's pass an id and load the data from API.

Reply:
I kind of agree with you, having 100% static page makes it really fast but I don't see this approach best practice. It's not that bad as well as the email goes straight to that user and the data is no that sensitive, but at the very least we should avoid doing something like this even if we won't have a complete static page.